File: api/src/main/java/net/blueshell/api/dto/CommitteeMemberDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.dto.user.SimpleUserDTO;

@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "CommitteeMember")
public class CommitteeMemberDTO extends BaseDTO {
    private Long id;
    @NotBlank
    private Long userId;
    @NotNull
    private Long committeeId;
    @NotBlank
    private String role;
    private SimpleUserDTO user;
}

File: api/src/main/kotlin/net/blueshell/api/committee/dto/CommitteeMemberDTO.kt
package net.blueshell.api.committee.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import net.blueshell.api.shared.dto.AuditedSoftDeleteDTO

@Schema(name = "CommitteeMember")
data class CommitteeMemberDTO(
    @field:NotNull
    var userId: Long? = null,

    @field:NotNull
    var committeeId: Long? = null,

    @field:NotBlank
    var role: String? = null
) : AuditedSoftDeleteDTO()
